cli/command/stack/deploy_composefile.go

```diff
@@ -320,6 +322,13 @@ func deployServices(
 			if sendAuth {
 				updateOpts.EncodedRegistryAuth = encodedAuth
 			}
+
+			if image != service.Spec.TaskTemplate.ContainerSpec.Image {
```

Comment:
That sounds reasonable to me.

We would need to make sure there is a way to enable digest pinning if the service was originally deployed with `--no-resolve-image`, or vice versa.